src/coreclr/tools/aot/ILCompiler/ILCompiler.csproj

```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <OutputPath>$(RuntimeBinDir)ilc/</OutputPath>
-    <RuntimeIdentifier>$(OutputRid)</RuntimeIdentifier>
+    <RuntimeIdentifier>$(PackageRID)</RuntimeIdentifier>
```

Comment:
It gets packed by a special project file pkgproj `src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj`, which is ultimately using `$(OutputRid)` (via properties inheritance).

This change is correct. The reason we haven't gotten any complaints before is because we are not (yet) cross-compile ILCompiler  in a way that HostOS and TargetOS differ (the first part of RID...).

Reply:
@am11 does `Microsoft.DotNet.ILCompiler.pkgproj` consume files built by `ILCompiler.csproj`?
If so, for what rid were these files built? `OutputRid` or `PackageRID`?

Reply:
> `Microsoft.DotNet.ILCompiler.pkgproj` consume files built by `ILCompiler.csproj`

It collects the files from various location and packs them together in a zip (.nupkg) file. In this case, `PackageRID` is only used to restore the packages when building the managed assembly (we don't have proeprocessor conditions based on RuntimeIdentifier, it's all MSIL or `#if TARGET_{OS_or_ARCH}` in the code). OutputRid is used in pkgprojs or passed to cmake to decide which bits to build and pack.

Reply:
Sounds like we won't break anything then.

From what you said, I wonder: does `ILCompiler.csproj` need a `RuntimeIdentifier`?

Reply:
> does `ILCompiler.csproj` need a `RuntimeIdentifier`?

Good question. As I understood it; `RuntimeIdentifer` during the restore step is used to decide which `runtime.{rid}.xyz` for a given package to restore, if the package has native assets. Here, ILCompiler.props has multiple PackageReference and ProjectReference, it is likely that the resolved dependency graph of this project (under `obj` directory) has such `package-with-native-assets` which is requiring the `RuntimeIdentifier` to be specified.